packages/xml-lsp-server/src/serverHelper.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  TextDocument,
+  Diagnostic,
+  DiagnosticSeverity,
+  Range
+} from "vscode-languageserver";
+
+import { ILexingError, IRecognitionException } from "chevrotain";
+import { parse } from "@xml-tools/parser";
+
+export const SYNTAX_ERROR_MSG = "Syntax error";
+
+const lexingErrorToDiagnostic = (document: TextDocument) => (
+  error: ILexingError
+): Diagnostic => ({
+  message: error.message,
+  range: Range.create(
+    document.positionAt(error.offset),
+    document.positionAt(error.offset + error.length)
+  ),
+  severity: DiagnosticSeverity.Error,
+  source: SYNTAX_ERROR_MSG
+});
+
+const parsingErrorToDiagnostic = (document: TextDocument) => (
```

Comment:
`parsingErrorToDiagnostic` is a function, declare it directly as such
no need to assign a function Object to a const variable.

Reply:
It's a map method, that maps a nested object, what's wrong with it being declared in this way?

Reply:
- **Consistency**, why are some functions using `function` **keyword** and others are assigned to a variable.
- **Different runtime behavior:**
  - regular functions gets **hoisted** to the top of the file and can be invoked before their definition or even in a cyclic manner, sort of pre-processing of the js file phase.
  - functions references inside variables are defined at actual runtime phase and cannot access not yet defined other function references in variables.
  - The **fat arrow** functions actual also bind the "this" context:
    - for **fat arrow** it would be an empty object (but can be modified)
    - for regular functions it would be a **global** object.

```javascript
const foo = () => {
  bar();
};

// won't work (comment in to check)
// foo();

const bar = () => {
  console.log("hi")
};

// will work
foo();


function foo2() {
  bar2();
}

// will work
foo2();

function bar2() {
  console.log("hello")
}
```

```javascript
function foo(){
  console.log(this) // inspect `this` in debugger
}
foo();

var foo2 = () =>{
  console.log(this) // inspect `this` in debugger
};

foo2();

```

So basically we add complexity that requires in-depth knowledge of JS runtime to understand and can cause some potential class of bugs without any benefit or gain...

Such style of functions declarations is used when dynamically creating functions.
e.g: https://lodash.com/docs/4.17.15#partial